### PR TITLE
[VEN-248] Wire up withdraw buttons on non vesting vaults (VAI and VRT)

### PR DESCRIPTION
**Pull request by therealemjy**
## Jira ticket(s)

VEN-248

## Changes

- add `useHandleTransactionMutation` hook to handle the most common transaction interactions, where a success transaction modal is displayed on success and a toast is displayed on failure
- add modal to withdraw some of the staked tokens in VAI vault
- wire up withdraw button of VRT vault to withdraw all the staked tokens at once (the contracts don't offer the possibility to withdraw a certain amount, this is also the current UX in production)

### CODE REVIEW

File: src/hooks/useHandleTransactionMutation.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import { toast } from 'components/v2/Toast';
+
+export interface IHandleMutationInput {
+  mutate: () => Promise<TransactionReceipt | void>;
+  successTransactionModalProps: (
+    transactionReceipt: TransactionReceipt,
+  ) => OpenSuccessfulTransactionModalInput;
+}
+
+const useHandleTransactionMutation = () => {
```
Comment (coreyar) [inline src/hooks/useHandleTransactionMutation.ts:16]: Nice

File: src/pages/Dashboard/MintRepayVai/MintVai/index.spec.tsx [hunk 2]
```diff
@@ -123,7 +123,7 @@ describe('pages/Dashboard/MintRepayVai/MintVai', () => {
     expect(openSuccessfulTransactionModal).toHaveBeenCalledWith({
       transactionHash: fakeTransactionReceipt.transactionHash,
       amount: {
-        tokenId: 'xvs',
+        tokenId: 'vai',
```
Comment (coreyar) [inline src/pages/Dashboard/MintRepayVai/MintVai/index.spec.tsx]: I don't have a preference but on my PR it was suggested we grab this from the Tokens constants

File: src/pages/Dashboard/MintRepayVai/RepayVai/index.spec.tsx [hunk 3]
```diff
@@ -116,7 +116,7 @@ describe('pages/Dashboard/MintRepayVai/RepayVai', () => {
     expect(openSuccessfulTransactionModal).toHaveBeenCalledWith({
       transactionHash: fakeTransactionReceipt.transactionHash,
       amount: {
-        tokenId: 'xvs',
+        tokenId: 'vai',
```
Comment (coreyar) [inline src/pages/Dashboard/MintRepayVai/RepayVai/index.spec.tsx:119]: should we get this from token constants?

**Review by coreyar — APPROVED** [review]

File: src/pages/Dashboard/MintRepayVai/MintVai/index.spec.tsx (hunk 2 above)
Comment (therealemjy) [reply src/pages/Dashboard/MintRepayVai/MintVai/index.spec.tsx]: This is a test, I think it's OK to have hard coded values in this case. But let's grab it from the constant too, this way we're consistent across the board

File: src/pages/Dashboard/MintRepayVai/RepayVai/index.spec.tsx (hunk 3 above)
Comment (therealemjy) [reply src/pages/Dashboard/MintRepayVai/RepayVai/index.spec.tsx:119]: Yeah sure

**Pull request merged**